### DISCUSSION TITLE
Idea for better handling errors with Ignition

### DISCUSSION THREAD

**Discussion by empiricompany** [OpenMage/magento-lts#3955] [Ideas]
Thanks @fballiano for your smart PR about ignition support, i think it has great potential.

I was looking into how to manage the configuration file; for better integration through an external extension, we could add these two observers:

![new_events](https://github.com/OpenMage/magento-lts/assets/5071467/9e4a1180-d299-4956-bb3d-97355aa5695c)

This way, we can implement it better, taking inspiration from the Drupal module:
https://git.drupalcode.org/project/ignition/tree/1.0.x

what do you think about?

PS: I'm quite busy at work right now, so I wouldn't be able to work on it immediately, but I wanted to do some brainstorming in the meantime

**Comment by empiricompany** [comment]
It would be amazing to also leverage the integration with Flare and with OpenAI!
Also, for "branding/marketing", being listed among the supported extensions with OpenMage would give us a nice boost.

**Reply by fballiano** [reply]
I thought about it but I wanted to implement that minimum modification to the core that was necessary and couldn't be (as far as I know) added via a module.
maybe some more feature can be added without having to modify the core? maybe with the 2 events it would be possible to have an ignition module instead of the patch to the core?

**Reply by empiricompany** [reply]
Yes, I was thinking exactly that. By adding those two events, the module can observe into and manage everything better. 
But as I mentioned, for now, I would leave your patch because I don't have time to implement it.

**Comment by luigifab** [comment]
This [PR](https://github.com/spatie/ignition/pull/262) for ignition looks like #3679 🤫

**Reply by fballiano** [reply]
yep, ignition supports linking to the IDE, it works fine for me

**Comment by fballiano** [comment]
I wanted to use an event to trigger ignition but it wasn't available so I added it to the core :-)
But thinking about it, if we add the dispatchevent and trigger Ignition from that event, we're adding a few calls to the stack, which will make the output of ignition a bit harder to read since you will have to go back 2-3 steps in the call stack to see the code that actually broke.
It's worth testing tho.

**Reply by empiricompany** [reply]
True, I hadn't figured that out. It would also show in the stack, you mean this, right?

**Comment by empiricompany** [comment]
@fballiano i have started here:
https://github.com/empiricompany/openmage_ignition

seems ok, to handle it we have to add 2/3 new events:
`Mage.php`
mage_print_exception_before
mage_print_exception_after (not used by the extension)
`App.php`
core_app_init_before

**Comment by sreichel** [comment]
Great idea and another reason to make php-8.1 required :+1:
